File: server/Models/Video.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using server.Models;

namespace Models
{
    public class Video
    {
        public string VideoId { get; set; } = null!;
        public string VideoUrl { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime UploadDate { get; set; } = DateTime.UtcNow;

        // Navigation Properties
        public string StudentId { get; set; }
        public Student Student { get; set; }
        public List<Comment> Comments { get; set; }
        public List<Rating> Ratings { get; set; }
    }
}

File: client/src/models/Video.ts
import { Student } from './Student';
import { Comment } from './Comment';
import { Rating } from './Rating';

export interface Video {
    videoId: number;
    videoUrl: string;
    title: string;
    description: string;
    uploadDate: string;
    studentId: string; // Reference to the student who uploaded the video
    student?: Student;
    comments?: Comment[]; // Comments on the video
    ratings?: Rating[]; // Ratings for the video
}
